cli/graph_util.rs

```diff
@@ -320,8 +320,40 @@ impl ModuleGraphBuilder {
       self.resolver.force_top_level_package_json_install().await?;
     }
 
+    // add the lockfile redirects to the graph if it's the first time executing
+    if graph.redirects.is_empty() {
+      if let Some(lockfile) = &self.lockfile {
+        let lockfile = lockfile.lock();
+        for (from, to) in &lockfile.content.redirects {
+          if let Ok(from) = ModuleSpecifier::parse(from) {
+            if let Ok(to) = ModuleSpecifier::parse(to) {
+              if !matches!(from.scheme(), "file" | "npm")
+                && !matches!(to.scheme(), "file" | "npm")
```

Comment:
I appreciate your effort.
Though the amount of nesting is relatively high. you are on level 7 here. 
Maybe you can avoid some nesting by combining if's and pulling the contents of the for-loop into a small function to increase readability.